content/en/docs/reference/kubectl/conventions.md

```diff
@@ -36,42 +36,56 @@ For `kubectl run` to satisfy infrastructure as code:
+| ReplicationController _(deprecated)_ | v1                 | `kubectl run --generator=run/v1`                  |
+| Deployment _(deprecated)_            | extensions/v1beta1 | `kubectl run --generator=deployment/v1beta1`      |
+| Deployment _(deprecated)_            | apps/v1beta1       | `kubectl run --generator=deployment/apps.v1beta1` |
+| Job _(deprecated)_                   | batch/v1           | `kubectl run --generator=job/v1`                  |
+| CronJob _(deprecated)_               | batch/v2alpha1     | `kubectl run --generator=cronjob/v2alpha1`        |
+| CronJob _(deprecated)_               | batch/v1beta1      | `kubectl run --generator=cronjob/v1beta1`         |
+{{< /table >}}
+
 
 {{< note >}}
 Generators other than `run-pod/v1` are deprecated.
 {{< /note >}}
 
-If you do not specify a generator flag, other flags prompt you to use a specific generator. The following table lists the flags that force you to use specific generators, depending on the version of the cluster:
+If you do not explicitly specify a generator when you invoke `kubectl run`, a generator can be implicitly set based on the flags you chose. The following table lists flags and the generators that are activated:
 
+{{< table caption="kubectl run flags and the resource they imply" >}}
 |   Generated Resource  | Flag                    |
 |:---------------------:|-------------------------|
 | CronJob               | `--schedule=<schedule>` |
 | Deployment            | `--restart=Always`      |
 | Job                   | `--restart=OnFailure`   |
 | Pod                   | `--restart=Never`       |
+{{< /table >}}
 
 {{< note >}}
 These flags use a default generator only when you have not specified any flag.
-This means that when you combine `--generator` with other flags the generator that you specified later does not change. For example, in a cluster v1.4, if you initially specify
-`--restart=Always`, a Deployment is created; if you later specify `--restart=Always`
-and `--generator=run/v1`, a ReplicationController is created.
-This enables you to pin to a specific behavior with the generator,
-even when the default generator is changed later.
+This means that when you combine `--generator` with other flags, the generator that you specified later does not change.
 {{< /note >}}
 
-The flags set the generator in the following order: first the `--schedule` flag, then the `--restart` policy flag, and finally the `--generator` flag.
+For example: you write a script that, initially, invokes:
+```bash
+kubectl run --restart=Always
+```
+The script creates a Deployment when you run it.
+
+Later, you decide to change the script to invoke
+```bash
+kubectl run --restart=Always --generator=run/v1
+```
+Now, when you run the script, it creates a ReplicationController because you specified the (deprecated) generator that you want kubectl to use.
+
+If you don't specify a generator, kubectl pays attention to other flags in the following order:
+1. `--schedule`
```

Comment:
@sftim. Great work! 😀  I don't want to drag this out; sorry if this is taking up a lot of time.
nit, could fix later: The original sentence covers all the flags. How about adding back the --generator flag as the last flag to be processed?
nit fix now: If you use a list, add a new line at line 85.
Kubectl processes the flags in the following order:

1. ...

Reply:
I had another go.

If this doesn't land via the v1.17 release, I can cherry-pick the same changes against _master_.

Reply:
@sftim. Last thing: Add a new line before line 67. The list does not _render_. Thanks!